### ISSUE TITLE
InCoder: A Generative Model for Code Infilling and Synthesis

### ISSUE THREAD

**Issue by InabaTatsuro** [InabaTatsuro/papers#32]
# InCoder: A Generative Model for Code Infilling and Synthesis [[ICLR23](https://arxiv.org/abs/2204.05999)]

![スクリーンショット 2023-11-22 17 48 40](https://github.com/InabaTatsuro/papers/assets/102784221/7d75b784-1381-477a-bf57-d7a8280e20c1)

## Abstract
- InCoder can infill the program via left-to-right generation
- train by maximizing logP([left; <mask>; right; <mask>; span; <EOM>]) and inference by sampling tokens autoregressively from the distributions P(・| [left; <mask>; right; <mask>])

## Result
![スクリーンショット 2023-11-22 18 11 42](https://github.com/InabaTatsuro/papers/assets/102784221/4d05aad5-198e-4cb9-a78b-2ed42d9afea9)

- showed higher performance than other pre-trained models in Human eval

![スクリーンショット 2023-11-22 18 12 55](https://github.com/InabaTatsuro/papers/assets/102784221/717cfdfe-61b1-4b17-a6f8-469dc2d1c26a)

- The proposed method is Zero-shot setting, whereas the bottom settings are finetuned with 250K samples
